addons/purchase_stock/models/stock_move.py

```diff
@@ -186,10 +186,13 @@ def _get_value_from_quotation(self, quantity, at_date=None):
             return super()._get_value_from_quotation(quantity, at_date)
         price_unit = self.purchase_line_id._get_stock_move_price_unit()
         quantity = min(quantity, self.quantity)
```

Comment:
As `quantity` comes from `_get_valued_qty()`, which now should always give the quantity in the uom of the product, we need to cast `self.quantity` to the uom of the product.